src/Catty/Parser&Serialializer/New Parser & Serializer/Parser/Context/CBXMLParserContext.h

```diff
@@ -34,20 +34,19 @@
 
 @property (nonatomic, readonly) CGFloat languageVersion;
 
+@property (nonatomic, strong) GDataXMLElement *rootElement;
+
 //------------------------------------------------------------------------------------------------------------
-// ressources data used while traversing the tree
+// resources data used while traversing the tree
 //------------------------------------------------------------------------------------------------------------
-// TODO: refactor this later: remove brickList here and dynamically find brick in scriptList. maybe scripts should be referenced in bricks as well!!
 @property (nonatomic, strong) NSMutableArray *programVariableList; // (used for parsing only)
 @property (nonatomic, strong) NSMutableArray *programListOfLists; // (used for parsing only)
 
-@property (nonatomic, strong) NSMutableDictionary *spriteObjectNameVariableList; // (used for parsing only)
-@property (nonatomic, strong) NSMutableDictionary *spriteObjectNameListOfLists; // (used for parsing only)
-@property (nonatomic, strong) NSMutableDictionary *formulaVariableNameList; // (used for parsing only)
-@property (nonatomic, strong) NSMutableDictionary *formulaListNameList; // (used for parsing only)
 @property (nonatomic, strong) NSMutableSet<NSString*> *unsupportedElements; // (used for parsing only)
 
-- (id)initWithLanguageVersion:(CGFloat)languageVersion;
+- (instancetype)initWithLanguageVersion:(CGFloat)languageVersion andRootElement:(GDataXMLElement *)rootElement NS_DESIGNATED_INITIALIZER;
+- (instancetype)init NS_UNAVAILABLE;
```

Comment:
Since `new` is not unavailable yet, this class can still be initialized without a language version or root element.